src/orion/service/client/experiment.py

```diff
@@ -0,0 +1,210 @@
+            strategy=strategy,
+            max_trials=max_trials,
+            max_broken=max_broken,
+            branching=branching,
+            max_idle_time=max_idle_time,
+            heartbeat=heartbeat,
+            working_dir=working_dir,
+            debug=debug,
+        )
+
+        client = ExperimentClientREST(
+            experiment,
+            executor=executor,
+            heartbeat=heartbeat,
+        )
+
+        storage = ClientActionREST(experiment, endpoint, token)
+
+        client.workon_client = workon
+        client.storage_client = storage
+        return client
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, experiment, executor=None, heartbeat=None):
+        # Do not call super here; we do not want to instantiate the producer
+        self.workon_client = None
+
+        if heartbeat is None:
+            heartbeat = orion.core.config.worker.heartbeat
+
+        self._experiment = experiment
+        self.heartbeat = heartbeat
+        self._executor = executor
+        self._executor_owner = False
+        self.storage_client = None
+        self.plot = PlotAccessor(self)
+
+    def to_pandas(self, with_evc_tree=False):
+        raise RuntimeError()
+
+    @property
+    def _pacemakers(self):
+        # pylint: disable=protected-access
+        return self.workon_client._pacemakers
+
+    #
+    # Workon REST API overrides
+    #
+
+    @property
+    def is_broken(self):
+        """See `~ExperimentClient.is_broken`"""
+        try:
+            self.is_done
+        except BrokenExperiment:
+            return True
+        return False
+
+    @property
+    def is_done(self):
+        """See `~ExperimentClient.is_done`"""
+        return self.workon_client.is_done()
+
+    def suggest(self, pool_size=0) -> TrialCM:
+        """See `~ExperimentClient.suggest`"""
+        remote_trial = self.workon_client.suggest(pool_size=pool_size)
+        return remote_trial
+
+    def observe(self, trial, results, name: str = "objective"):
+        """See `~ExperimentClient.observe`"""
+        self.workon_client.observe(trial, results, name=name)
+
+    def release(self, trial, status="interrupted"):
+        """See `~ExperimentClient.release`"""
+        # we probably should not expose this
+        return self.workon_client.release(trial, status)
+
+    #
+    # Disabled
+    #
+
+    def reserve(self, trial):
+        """See `~ExperimentClient.reserve`"""
+        raise NotImplementedError("REST API reserve on suggest")
+
+    @property
+    def producer(self):
+        """Return the producer configuration of the experiment."""
+        raise NotImplementedError("REST API does not instantiate the producer")
+
+    @property
+    def storage(self):
+        """Return the storage currently in use by this client"""
+        raise NotImplementedError("REST API does not have a storage")
+
+    #
+    # Storage REST API
+    #
+
+    def insert(self, params, results=None, reserve=False):
```

Comment:
It's not great that insert behaves differently. I did not look at the workon module but perhaps that is where insert should be, as making `insert(params, results=None, reserve=True)` is a use-case for working on a trial.

Reply:
Isn't the algo running on the server side generating the parameters ?
Workon is only fetching+reserving the trials.

The only use case for inserting from the API is if a user wants to manually add a trial to be worked on by which ever worker is available ?